opm/parser/eclipse/EclipseState/SummaryConfig/SummaryConfig.cpp

```diff
@@ -227,7 +227,19 @@ namespace Opm {
 
         using namespace std::placeholders;
         const auto handler = std::bind( handleKW, _1, schedule, props, n_xyz );
+
+#ifdef _MSC_VER
+        for (auto& x : section)
+        {
+            auto keywords = handler(x);
+            for (auto& keyword : keywords)
+            {
+                this->keywords.push_back(keyword);
+            }
+        }
+#else
         this->keywords = fun::concat( fun::map( handler, section ) );
```

Comment:
> It would most certainly be more lines, I don't know where you're getting that from. And it's not like you don't need to know the interfaces you use anyway, so that's not an argument.

```c++
for (auto& x : section)   {
    for (auto& keyword : handleKW(x, schedule, props, n_xyz))
       this->keywords.push_back(keyword);
}
```

vs 

```c++
        using namespace std::placeholders;
         const auto handler = std::bind( handleKW, _1, schedule, props, n_xyz );
         this->keywords = fun::concat( fun::map( handler, section ) );
```

I'd call this a draw. on the other hand the second version requires more implicit knowledge.

Let's stop the discussion here, I'm getting tired of it: If you want to use functional style programming, that's fine with me, even though for maintenance reasons I advice to avoid it if it does not come with clear benefits other than "it is more elegant".